File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
class Solution {
    public int[] solution(int[] arr, int n) {
        int[] answer = {};
        boolean odd_even = arr.length % 2 == 0 ? true : false;
        
        for(int i = 0; i < arr.length; i++){
            if(odd_even) i++;
            arr[i] += n;
            if(!odd_even) i++;
        }
        
        return arr;
    }
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    answer = []
    
    if len(arr) % 2:
        answer = [x + n if i % 2 == 0 else x for i, x in enumerate(arr)]
    else:
        answer = [x + n if i % 2 == 1 else x for i, x in enumerate(arr)]
        
    return answer